### PR TITLE
Modest Memory Improvements

### PR DESCRIPTION
**Pull request by mprahl**
See each commit for more details.

### CODE REVIEW

**Comment by yiraeChristineKim** [comment]
good good curious Just how much this is improved compared to previous

File: main.go [hunk 1]
```diff
@@ -232,6 +232,27 @@ func main() {
 				&corev1.Secret{}: {
 					Field: fields.SelectorFromSet(fields.Set{"metadata.name": propagatorctrl.EncryptionKeySecret}),
 				},
+				&clusterv1.ManagedCluster{}: {
+					Transform: func(obj interface{}) (interface{}, error) {
+						cluster := obj.(*clusterv1.ManagedCluster)
+						// All that ManagedCluster objects are used for is to check their existence to see if a
+						// namespace is a cluster namespace.
+						guttedCluster := &clusterv1.ManagedCluster{}
+						guttedCluster.SetName(cluster.Name)
```
Comment (yiraeChristineKim) [inline main.go:241]: Label not used? I found this `ManagedClusterLabels map[string]string`

**Comment by mprahl** [comment]
> good good curious Just how much this is improved compared to previous

For real results, we'll have to wait for a performance and scale test but in my fake one, it does reduce the memory allocations by several GB but it doesn't move the needle much on actual memory in use after it becomes idle.

File: main.go (hunk 1 above)
Comment (mprahl) [reply main.go:241]: It's a bit confusing but that ManagedCluster isn't queried by the controller-runtime caching client but it's coming from the policy templating client (which uses the caching dependency watcher library):
https://github.com/open-cluster-management-io/governance-policy-propagator/blob/a06836bba155f9ca30edd0ded268d38d222baf3f/controllers/propagator/propagation.go#L285

**Review by JustinKuli — APPROVED** [review]
Three nice improvements!

**Pull request merged**